src/Core/Containers/VariableSet.hpp

```diff
@@ -613,13 +675,25 @@ auto VariableSet::insertVariable( const std::string& name, const T& value )
 }
 
 template <typename T>
-auto VariableSet::getVariable( const std::string& name ) const -> T& {
-    assert( existsVariable<T>( name ) );
+auto VariableSet::getVariable( const std::string& name ) -> T& {
+    return const_cast<T&>( const_cast<const VariableSet*>( this )->getVariable<T>( name ) );
+}
+
+template <typename T>
+auto VariableSet::getVariable( const std::string& name ) const -> const T& {
+
+    if constexpr ( std::is_enum<T>::value ) {
```

Comment:
This test should not be on this condition but on the existance of an enumConverter, then should not be constexpr. 
Testing on `is_enum` impose to store enumeration variable as values of the enumeration's underlying type and not as  typed enumeration. So, this propagates GLSL limitations about enumerations to VariableSet and I think this is not a good thing.
Enum management through EnumConverters, and the storage using underlying type is welcome for several operations (GLSL binding, edition gui, ... ) but should not be imposed as user should store enumeration without a converter and with the real enumeration type.

Reply:
Ok, I'll think of an alternative.
A first guess is to store the enums directly in the VariableSet (always), and have a dynamic visitor that convert to unerlying types in the ShaderParameters class.
I let the conversation open until a solution is found.

Reply:
(I close the other related conversations to have a simpler workflow)

Reply:
> Enum management depends on the usage of the enum, not on the VariableSet management. I think we do not have to impose some restrictions on the variable stored on VariableSet. We should definitly discuss about enum management as an integrated VariableSet service ...
> Why considering enum variable only through the enum underlying type ?

Reply:
new proposal
one can add/set/get plain enum with [insert/set/get}Variable
or set get underlying with [set/get]EnumVariable, this also support string setters thru enumConverters.
TODO
maybe change the name of the second set of method, and also add enumConverters to plain enum type.
add enable_if to ensure template type compatibility

Reply:
Another solution is to move enum management to funtion, in a specific namespace, e.g. `VariableSetEnumManagement`

Thu
`vs.setEnumVariable(...)` becomes `VariableSetEnumManagement::setEnumVariable(vs, ...)`

Reply:
I prefer this way that should allow both adding/retrieving an enum through its real type or through its underlying type but also, as you mentionned, to envision other helpers to extend VariableSetManagement without modifying VariableSet.